### PR TITLE
Add missing documentation and minor typos fix

### PR DESCRIPTION
**Pull request by fvisin**
This PR adds every non-documented class to the autobuilt documentation and fixes a few typos.
Please check if you like the way I grouped things together.

### CODE REVIEW

File: docs/development/internal_api.rst [hunk 1]
```diff
@@ -12,3 +20,73 @@ Internal API
...
+
+.. automodule:: blocks.monitoring.evaluators
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Search
+------
+.. automodule:: blocks.search
```
Comment (bartvm) [inline docs/development/internal_api.rst]: Some of these modules shouldn't go into the internal API. The idea is that the internal API is used to document modules and objects that a user would normally not interface with (but a developer would) e.g. `ApplicationCall`, `BoundApplication`, that sort of things. Beam search, the wrapper bricks, and some of the methods in the aggregation module should be public though.

File: docs/development/internal_api.rst [hunk 2]
```diff
@@ -12,3 +20,73 @@ Internal API
...
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Utils
+-----
+.. automodule:: blocks.config_parser
```
Comment (bartvm) [inline docs/development/internal_api.rst]: This docstring is rendered at http://blocks.readthedocs.org/en/latest/configuration.html already

Comment (fvisin) [inline docs/development/internal_api.rst]: I missed it, sorry!

File: docs/development/internal_api.rst (hunk 1 above)
Comment (fvisin) [inline docs/development/internal_api.rst]: I should have corrected. Let me know if it is ok before merging please and I will rebase.

**Comment by fvisin** [comment]
Should be ready to merge, let me know if anything is not where it should be.

**Pull request merged**